R/ValidationFunctions.R

```diff
@@ -0,0 +1,290 @@
+#' Validate result based on specified tolerance
+#' @param result A data object to be validated
+#' @param abs_diff A logical value indicating whether to validate absolute values
+#' @param tolerance A numeric value setting tolerance of the comparison
+#' @return A list contains confrontation details and validation results
+validateResult <- function(result, abs_diff = TRUE, tolerance) {
+  result[is.na(result)] <- 0
+  # Validate result
+  if (abs_diff) {
+    validation <- as.data.frame(abs(result) <= tolerance)
+  } else {
+    validation <- as.data.frame(result <= tolerance)
+  }
```

Comment:
@WesIngwersen , these comparisons changed from `<` to `<=` which is having ramifications where the tolerance is 0. (i.e., creating many more failures since it's hitting the tolerance).

Could we switch it back to `<`? Or would you prefer special handling for `tolerance=0`

Reply:
I looked at git blame and don't see where they were `<` before.  
however I don't have any issue with your suggested change except we need to be clear in docs/comments that tolerance cannot be set to 0 in any case.